src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -143,26 +134,24 @@ func newCommitLogReader(opts Options, seriesPredicate SeriesFilterPredicate) com
 		chunkReader:       newChunkReader(opts.FlushSize()),
 		infoDecoder:       msgpack.NewDecoder(decodingOpts),
 		infoDecoderStream: msgpack.NewDecoderStream(nil),
-		decoderQueues:     decoderQueues,
-		decoderBufPools:   decoderBufs,
-		outChan:           outBuf,
-		cancelCtx:         cancelCtx,
-		cancelFunc:        cancelFunc,
-		shutdownCh:        make(chan error),
-		metadata:          readerMetadata{},
+		completedDecoders: completedDecoders{wg: &sync.WaitGroup{}},
```

Comment:
Do you always initialize a waitgroup?  If so can it just be an embedded struct?  Do you need the nil-ability, etc?